src/modules/launcher/Plugins/Microsoft.Plugin.Program/Main.cs

```diff
@@ -60,22 +68,35 @@ public void Save()
 
         public List<Result> Query(Query query)
         {
-            var results1 = _win32ProgramRepository.AsParallel()
-                .Where(p => p.Enabled)
-                .Select(p => p.Result(query.Search, _context.API));
+            foreach (var programArgumentParser in _programArgumentParsers)
+            {
+                if (!programArgumentParser.Enabled)
+                {
+                    continue;
+                }
+
+                if (!programArgumentParser.TryParse(query, out var program, out var programArguments))
+                {
+                    continue;
+                }
 
-            var results2 = _packageRepository.AsParallel()
-                .Where(p => p.Enabled)
-                .Select(p => p.Result(query.Search, _context.API));
+                var results1 = _win32ProgramRepository.AsParallel()
+                    .Where(p => p.Enabled)
+                    .Select(p => p.Result(program, programArguments, _context.API));
 
-            var result = results1.Concat(results2).Where(r => r != null && r.Score > 0);
-            if (result.Any())
-            {
-                var maxScore = result.Max(x => x.Score);
-                result = result.Where(x => x.Score > Settings.MinScoreThreshold * maxScore);
+                var results2 = _packageRepository.AsParallel()
+                    .Where(p => p.Enabled)
+                    .Select(p => p.Result(program, programArguments, _context.API));
+
+                var result = results1.Concat(results2).Where(r => r != null && r.Score > 0);
+                if (result.Any())
+                {
+                    var maxScore = result.Max(x => x.Score);
+                    result = result.Where(x => x.Score > Settings.MinScoreThreshold * maxScore);
+                }
             }
 
-            return result.ToList();
+            return new List<Result>(0);
```

Comment:
Thanks @royvou for this PR.  I apologize that I'm just taking a look now.   Doesn't this always return an empty list of programs?

Reply:
😭 Looks like in one of the merges I did not correctly merge/test it live. Unit tests were still passing (but those don't check the main program class.